src/main/java/core/basesyntax/strategy/MyFileWriterImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.strategy;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class MyFileWriterImpl implements MyFileWriter {
+    public void write(String report, String filePath) {
+        File file = new File(filePath);
+        File parentDir = file.getParentFile();
+        if (parentDir != null && !parentDir.exists()) {
+            parentDir.mkdirs();
+        }
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(file))) {
+            writer.write(report);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write data to" + filePath, e);
```

Comment:
The error message in the RuntimeException is missing a space between 'to' and the file path. It should be "Can't write data to " to ensure the exception message is readable.